video/out/wayland_common.c

```diff
@@ -3292,14 +3415,29 @@ bool vo_wayland_init(struct vo *vo)
     wl_display_roundtrip(wl->display);
 
 #if HAVE_WAYLAND_PROTOCOLS_1_41
-    // Only bind to vo_dmabuf_wayland for now to avoid conflicting with graphics drivers
-    if (wl->color_manager && wl->supports_parametric && !strcmp(wl->vo->driver->name, "dmabuf-wayland")) {
+    // Only bind color surface to vo_dmabuf_wayland for now to avoid conflicting with graphics drivers
+    if (wl->color_manager && wl->supports_parametric && !strcmp(wl->vo->driver->name, "dmabuf-wayland"))
         wl->color_surface = wp_color_manager_v1_get_surface(wl->color_manager, wl->callback_surface);
-    } else {
-        MP_VERBOSE(wl, "Compositor does not support parametric image descriptions!\n");
+
+    if (wl->color_manager && wl->supports_icc) {
+        wl->color_surface_feedback = wp_color_manager_v1_get_surface_feedback(wl->color_manager, wl->callback_surface);
+        wp_color_management_surface_feedback_v1_add_listener(wl->color_surface_feedback, &surface_feedback_listener, wl);
     }
 #endif
 
+    if (!wl->supports_parametric)
+        MP_VERBOSE(wl, "Compositor does not support parametric image descriptions!\n");
+
+    struct gl_video_opts *gl_opts = mp_get_config_group(NULL, vo->global, &gl_video_conf);
+    if (wl->supports_icc) {
+        // dumb workaround for avoiding -Wunused-function
+        get_compositor_icc_file(wl);
```

Comment:
Could you explain what you meant by this comment?

Reply:
Oh that's just for our CI since we build with -Werror. It's to remind myself later to eliminate the branching redundancy whenever later in the future we hard depend on a newer wayland protocols version.

Reply:
Alright, I was worried you meant that this call was redundant since the function would get called anyway from the preferred_changed callback.

SDL had a bug in that direction: https://github.com/libsdl-org/SDL/issues/12408